interp/constrintern.ml

```diff
@@ -1657,6 +1639,29 @@ let drop_notations_pattern (test_kind_top,test_kind_inner) genv env pat =
       end
     | _ -> CErrors.anomaly Pp.(str "Invalid return pattern from Notation.interp_prim_token_cases_pattern_expr."))) x
   in
+  let add_pars ?loc g syndef_pl pl =
+    (* Asymmetric compatibility mode *)
+    if get_asymmetric_patterns () then
+      let expl_pl =
+        if syndef_pl = [] then
+          let env = Global.env () in
+          let n = match g with
+            | GlobRef.ConstructRef (ind,_ as cstr) ->
+              let n = List.length pl in
+              if Int.equal n (Inductiveops.constructor_nrealargs env cstr) then
+                Inductiveops.inductive_nparams env ind
+              else if Int.equal n (Inductiveops.constructor_nrealdecls env cstr) then
+                Inductiveops.inductive_nparamdecls env ind
+              else
+                error_wrong_numarg_constructor ?loc env cstr
+                  (Inductiveops.constructor_nrealargs env cstr)
+            | _ -> 0 in
+          List.make n (DAst.make @@ RCPatAtom None)
+        else
+          syndef_pl in
```

Comment:
Could you avoid this multiline `in` style, I really have trouble reading it for whatever reason.

Reply:
OK